pylint/lint.py

```diff
@@ -55,6 +55,25 @@
 
 MANAGER = astroid.MANAGER
 
+# FIXME(CPC): move to astroid
+def _ast_from_string(data, filepath, modname):
+    if modname in MANAGER.astroid_cache and MANAGER.astroid_cache[modname].file == filepath:
+        return MANAGER.astroid_cache[modname]
+    from astroid.builder import AstroidBuilder
+    return AstroidBuilder(MANAGER).string_build(data, modname, filepath)
+
+from io import StringIO, TextIOWrapper
+if sys.version_info >= (3, 0):
+    def _read_stdin():
```

Comment:
For both versions, I think it should read the binary contents and then read the first 2 lines (still in binary mode) and try to match it with the regexp described in: https://www.python.org/dev/peps/pep-0263/ for the declared encoding.

On Python 3, if the encoding is not declared, utf-8 can be considered the default thought (on Python 2 it should go for ascii if it's not declared).